staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go

```diff
@@ -258,6 +259,16 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope *RequestScope, forceWatc
 			if timeout == 0 && minRequestTimeout > 0 {
 				timeout = time.Duration(float64(minRequestTimeout) * (rand.Float64() + 1.0))
 			}
+
+			var emptyVersionedList runtime.Object
```

Comment:
Can we move that down to watchHandler function?

Reply:
I don't think so because we need the internal version. I have just realised that the name of the variable could be better.

Reply:
actually, after realising this actually an internal version then my original idea of adding the annotation from the storage layer could work (https://github.com/kubernetes/kubernetes/pull/126199).
The encoder does the conversion and add the type data. Am I right ?

/cc @benluddy

Reply:
You would need to select the negotiated embedded encoder first (currently happening in `serveWatchHandler`) and either plumb the encoder or the encoded list bytes through to the storage layer. I don't think either approach is very satisfying, but it's worth noting that making the storage-layer list call dependent on response content negotiation will change observable API behaviors (e.g. a client might start seeing 406 instead of 404).